packages/tslint-config/src/rules/blueprintIconComponentsRule.ts

```diff
@@ -45,15 +45,19 @@ function walk(ctx: Lint.WalkContext<string>): void {
         if (ts.isJsxAttribute(node) && node.name.text === "icon") {
             const { initializer } = node;
             const option = ctx.options;
-            if (ts.isStringLiteral(initializer) && option === OPTION_COMPONENT) {
+            if (initializer === undefined) {
+                // no-op
+            } else if (ts.isStringLiteral(initializer) && option === OPTION_COMPONENT) {
                 // "tick" -> <TickIcon />
```

Comment:
I know this is existing code, but this is pretty confusing to me. we're doing _both_ replacements suggested here and on L53?

Reply:
to be fair, this rule isn't even valid as there are no icon components yet. it's disabled by default.